### PR TITLE
Logic/home search filter

### PR DESCRIPTION
**Pull request by Hiperultimate**
Added functionality to sorting features and fixed some issues outside home search filter.

### CODE REVIEW

File: client/src/components/search/searchBar/sortBy/SortBy.jsx [hunk 1]
```diff
@@ -13,17 +13,37 @@ function SortBy({SortByState, SortState }) {
...
+            onClick={onClickLowest}
+            style={
+              SortBy === "Lowest" ? { color: "white" } : { color: "black" }
+            }
+            className={
+              SortBy === "Lowest"
+                ? "sort-by-items sort-selected"
+                : "sort-by-items"
+            }
+          >
```
Comment (DamFlancoz) [inline client/src/components/search/searchBar/sortBy/SortBy.jsx:46]: Maybe color style can go in sort-selected

File: client/src/utils/contexts/MapContext.js [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+    }
+}
+// FIX LATE ADDRESS DISPLAY IN DOCTOR REGISTER PAGE
 export async function searchToAddressResults(searchAddress) {
     try {
-        const getAddressResult = await fetch(
-          `https://geocoder.ls.hereapi.com/search/6.2/geocode.json?languages=en-US&maxresults=4&searchtext=${searchAddress}&apiKey=${process.env.REACT_APP_HERE_API_KEY}`
-        );
-        return getAddressResult;
+        searchAddress = searchAddress.replaceAll(' ', '+');
```
Comment (DamFlancoz) [inline client/src/utils/contexts/MapContext.js:11]: Try finding a react/js function which encodes a string url safe. This depending on how handled at api might wrong results for special characters like comma, dot, colon etc. If it is working fine for special characters this is good.

File: client/src/utils/filters/filterDoctors.js [hunk 3]
```diff
@@ -1,62 +1,87 @@
...
-        if(!doctorSpecialization.includes(filterSpecialization[i])){
-            return false;
-        };
-    }
+function withinFee(feeValue, doctorFee) {
+  if (doctorFee <= feeValue[1] && doctorFee >= feeValue[0]) {
     return true;
+  } else {
+    return false;
+  }
```
Comment (DamFlancoz) [inline client/src/utils/filters/filterDoctors.js:8]: Shorten to `return doctorFee <= feeValue[1] && doctorFee >= feeValue[0]`.
Whenever returning a boolean with if-else, check if it is possible to write a direct return statement like this.

File: client/src/utils/filters/filterDoctors.js [hunk 4]
```diff
@@ -1,62 +1,87 @@
...
+    }
+    if (!withinFee(feeValue, filterDoc.consultationFee)) {
+      continue;
+    }
+    if (!hasSpecialization(specializations, filterDoc.specialization)) {
+      continue;
     }
+    if (!hasTreatment(treatments, filterDoc.treatments)) {
+      continue;
+    }
```
Comment (DamFlancoz) [inline client/src/utils/filters/filterDoctors.js:65]: Can make a single if statement spanning lines. Multiple branching if statements are hard to follow, grouping them by action like rejection here linearizes them a bit. To organize different criteria of rejection newlines can be used but usually not necessary.
Example,
```
if(!searchDoctor === "" || !filterDoc.doctorName.toLowerCase().includes(searchDoctor.toLowerCase())
    || !withinFee(feeValue, filterDoc.consultationFee) 
    || !hasSpecialization(specializations, filterDoc.specialization)
    || !hasTreatment(treatments, filterDoc.treatments)) {
      continue;
    }
```

**Review by DamFlancoz — CHANGES_REQUESTED** [review]
Code looks good. Added a couple of refactoring pointers and considerations.

File: client/src/utils/filters/filterDoctors.js (hunk 4 above)
Comment (Hiperultimate) [reply client/src/utils/filters/filterDoctors.js:65]: I see...

File: client/src/components/search/searchBar/sortBy/SortBy.jsx (hunk 1 above)
Comment (Hiperultimate) [reply client/src/components/search/searchBar/sortBy/SortBy.jsx:46]: I won't be able to check for variable changes then.
